### PR TITLE
Add 106, don't fancy import trio

### PR DESCRIPTION
**Pull request by jakkdl**
Uh, I'll just let you add this one to the changelog&readme, and increment version, with there being three simultaneous open pull requests.

### CODE REVIEW

File: tests/trio106.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import importlib
+
+import trio
+import trio as foo
+from trio import *  # noqa
+from trio import blah, open_file as foo  # noqa
```
Comment (Zac-HD) [inline tests/trio106.py:6]: Ooh, this confused me for a minute.  Note to self: add a block comment below - "Note that our tests exercise the *Visitor* classes, without going through the noqa filter later in flake8 - so these suppressions are picked up by our project-wide linter stack but *not* the tests."

File: flake8_trio.py [hunk 2]
```diff
@@ -278,3 +288,4 @@ def run(self) -> Generator[Tuple[int, int, str, Type[Any]], None, None]:
 TRIO100 = "TRIO100: {} context contains no checkpoints, add `await trio.sleep(0)`"
 TRIO101 = "TRIO101: yield inside a nursery or cancel scope is only safe when implementing a context manager - otherwise, it breaks exception handling"
 TRIO102 = "TRIO102: it's unsafe to await inside `finally:` unless you use a shielded cancel scope with a timeout"
+TRIO106 = "TRIO106: trio must be imported with `import trio` for the linter to work"
```
Comment (Zac-HD) [inline flake8_trio.py:291]: Also needs a README entry

**Comment by jakkdl** [comment]
While on it I resolved the above

File: tests/trio106.py (hunk 1 above)
Comment (jakkdl) [reply tests/trio106.py:6]: Yeah it confused me too - had to double-check that the flake8 filters got applied when ran as a plugin.

**Comment by Zac-HD** [comment]
OK, apparently pushing commits marks the PR as closed rather than merged 😕 

It's merged to me though!

**Comment by jakkdl** [comment]
Yeah that's some funky UI. It's marked as Closed, but it also says `Pull request successfully merged and closed`. But no matter

It looks like TRIO106 is missing from the readme though

**Comment by jakkdl** [comment]
It could be worth adding a test that checks that each error message is mentioned in all four of the readme, changelog, plugin file and test file.

**Comment by Zac-HD** [comment]
Sounds good, go for it!

**Comment by jakkdl** [comment]
Opened an issue :+1:

**Pull request closed without merging**